File: 단계별로 풀어보기/1차원 배열/2577/숫자의 개수.cpp
#include <iostream>
using namespace std;

int multiplyInput(int A, int B, int C) {
	return A * B * C;
}

int* analyzeResult(int* cnt, int result) {

	while(result / 10 != 0){
		cnt[result % 10]++;
		result = result / 10;			
		}
		
	cnt[result % 10]++;

	return cnt;
}

int printCntResult(int* cnt) {

	for (int i = 0; i < 10; i++) {
		cout << cnt[i] << "\n";
	}

	return 0;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int A, B, C;
	int cnt[10] = { 0 };
	cin >> A >> B >> C;

	printCntResult(analyzeResult(cnt,multiplyInput(A, B, C)));

}

File: 단계별로 풀어보기/1차원 배열/2577/숫자의 개수.py
import sys
from functools import reduce

def multiplyInput(nums):
    return reduce(lambda x, y: x*y,nums)

def analyzeResult(cnt, result):

    while(result//10 != 0):
        cnt[result%10]+=1
        result=result//10

    cnt[result%10]+=1

    return cnt

def printCnt(cnt):
    for i in range(10):
        print(cnt[i])

def init():
    nums=[]
    for i in range(3):
        nums.append(int(sys.stdin.readline()))
    cnt=[0 for i in range(10)]
    printCnt(analyzeResult(cnt, multiplyInput(nums)))
    
init()
